### ISSUE TITLE
Seeking advice about conditional non-radio fields appearing inside radiogroup

### ISSUE THREAD

**Issue by brennanyoung** [w3c/aria-practices#2576]
In the case of a radio button group where one of the radio buttons is "Other" (or similar), requiring additional text input, what is the best (most reliable and expressive) way to **associate the input field programmatically** with that "Other"? 

<img width="549" alt="radio button group where one radio button, labelled 'Other' enables a secondary text input field" src="https://user-images.githubusercontent.com/1430985/210357212-d7968d4d-a046-427b-99b3-cfc91060fd30.png">

Is it enough to simply rely on DOM adjacency, or would a relationship attribute be wise? Which one?

The spec for `radiogroup` forbids any operable children that do not have the `radio` role. In HTML5, `<fieldset>` would permit the group to contain a mix because the radio button group is expressed via attributes, rather than DOM structure - so, how should we do it with ARIA?

Some suggestions:
1.
```
radiogroup#rg
    radio "foo"
    radio "bar"
    radio#r3 "other..."
input aria-labelledby="rg"
```
2.
```
radiogroup#rg
    radio "foo"
    radio "bar"
    radio#r3 "other..."
input aria-labelledby="r3"
```
Is it advisable to wrap the radiogroup and associated input field in an **additional** group? Seems like this better expresses "Info and Relationships" but I wonder whether it has any real world effect.
```
group
    radiogroup#rg
        radio "foo"
        radio "bar"
        radio#r3 "other..."
    input
```
    
Note, a related topic has been [explored by the uk government's accessibility team](https://design-system.service.gov.uk/components/radios/), although their inquiry is into aria-expanded.  No guidance is offered about a preferred DOM structure and/or relationship attributes.

**Comment by JAWS-test** [comment]
Yeah, that's a little tricky. However, I don't think the ARIA specification says that radio must be in radiogroup and a radiogroup must contain only radio. In the ARIA specification there is only one SHOULD in each case. Therefore I assume that also the following is allowed:
* radio + input in group or
* radio + input in radiogroup

However, it is also not entirely clear whether this is actually allowed, so a more detailed explanation in the ARIA specification would be helpful.

The input needs of course a meaningful label. DOM adjacency is never sufficient. Labeling can be done by aria-labelledby or aria-label or in case of an HTML input also by label.

**Comment by brennanyoung** [comment]
An extra curl in the tail: We're seeing problems putting the field inside the radiogroup. In some cases, the keystrokes entered into the input field will focus the radiobutton whose name begins with the pressed key/character.

PLEASE offer guidance for this not uncommon pattern.
